File: JavaScript Basics/09.ExamPreparation/JoroTheRabbit.js
function Solve(args) {
    var input = args[0].split(", ").map(Number),
        bestPath = Number.MIN_VALUE,
        len = input.length;

    for (var startIndex = 0; startIndex < len; startIndex++) {
        for (var step = 1; step < len; step++) {
            var index = startIndex;
            var currentPath = 1;
            var next = 0;
            if ((index + step) >= len) {
                next = (index + step) - len;
            } else {
                next = (index + step);
            }

            while (next != startIndex && input[index] < input[next]) {
                currentPath++;
                index = next;
                if ((index + step) >= len) {
                    next = (index + step) - len;
                } else {
                    next = (index + step);
                }
            }

            if (bestPath < currentPath) {
                bestPath = currentPath;
            }

        }
    }
    return bestPath;
}

var args = ["1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 0"];
console.log(Solve(args));

File: C# Part II Homework/10.ExamPreparation/JoroTheRabbit/JoroTheRabbit.cs
using System;

namespace JoroTheRabbit
{
    class JoroTheRabbit
    {
        static void Main(string[] args)
        {
            string input = Console.ReadLine();
            string[] cleanString = input.Split(new char[] { ',', ' ' }, StringSplitOptions.RemoveEmptyEntries);

            int[] values = new int[cleanString.Length];
            int bestPath = int.MinValue;

            for (int i = 0; i < cleanString.Length; i++)
            {
                values[i] = int.Parse(cleanString[i]);
            }

            for (int startIndex = 0; startIndex < values.Length; startIndex++)
            {
                for (int step = 1; step < values.Length; step++)
                {
                    int index = startIndex;
                    int currentPath = 1;
                    int next = (index + step) % values.Length;

                    while (next != startIndex && values[index] < values[next])
                    {
                        currentPath++;
                        index = next;
                        next = (index + step) % values.Length;
                    }

                    if (bestPath < currentPath)
                    {
                        bestPath = currentPath;
                    }

                }
            }
            Console.WriteLine(bestPath);
        }
    }
}
